src/Microsoft.Azure.WebJobs.ServiceBus/EventHubs/EventHubListener.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Config;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using System.Threading;
+using Microsoft.ServiceBus.Messaging;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+using System.Globalization;
+using Microsoft.Azure.WebJobs.Host.Triggers;
+using Microsoft.Azure.WebJobs.Host.Listeners;
+using Microsoft.Azure.WebJobs.Host.Executors;
+
+namespace Microsoft.Azure.WebJobs.ServiceBus
+{
+    // Created from the EventHubTrigger attribute to listen on the EventHub. 
+    internal sealed class EventHubListener : IListener, IEventProcessor, IEventProcessorFactory
+    {
+        private ITriggeredFunctionExecutor _executor;
+        private readonly EventProcessorHost _eventListener;
+        private readonly bool _singleDispatch;
+        private readonly EventProcessorOptions _options;
+
+        public EventHubListener(ITriggeredFunctionExecutor executor, EventProcessorHost eventListener, EventProcessorOptions options, bool single)
+        {
+            this._executor = executor;
+            this._eventListener = eventListener;
+            this._singleDispatch = single;
+            this._options = options;
+        }
+
+        public void Cancel()
+        {
+        }
+
+        public void Dispose()
+        {
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {            
+            return _eventListener.RegisterEventProcessorFactoryAsync(this, _options);
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            return _eventListener.UnregisterEventProcessorAsync();
+        }
+
+
+        async Task IEventProcessor.OpenAsync(PartitionContext context)
+        {
+            // Begin listener 
+        }
+
+        async Task IEventProcessor.ProcessEventsAsync(PartitionContext context, IEnumerable<EventData> messages)
+        {
+            EventHubTriggerInput value = new EventHubTriggerInput
+            {
+                _events = messages.ToArray(),
+                _context = context
+            };
+
+            // Single dispatch 
```

Comment:
What does this single dispatch mean for Dashboard? Is each function invocation then its own entry in the Dashboard?